### PR TITLE
bgpv1: Add GetRoutes method to Router interface and generic Path type

### PR DESCRIPTION
**Pull request by rastislavs**
Adds `GetRoutes` method to the Router interface and generic `Path` and `Route` types, that can be used to retrieve existing routes from the RIB of underlying router.

This is the first step towards exposing the BGP routes via cilium API + CLI.

### CODE REVIEW

File: pkg/bgpv1/gobgp/conversions_test.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+			res, err := s.AddPath(context.TODO(), &gobgp.AddPathRequest{
+				Path: path,
+			})
+			require.NoError(t, err)
+			require.NotZero(t, res.Uuid)
+		})
+	}
+}
+
+// TODO: Test ToAgentPath
```
Comment (YutaroHayakawa) [inline pkg/bgpv1/gobgp/conversions_test.go]: Could you address this TODO?

**Review by YutaroHayakawa — CHANGES_REQUESTED** [review]
Thanks for taking over this! Overall looks good except some minor things. Could you consider consolidating last two commits into first two? So that we will have a clean commit history.

File: pkg/bgpv1/gobgp/conversions_test.go (hunk 1 above)
Comment (rastislavs) [reply pkg/bgpv1/gobgp/conversions_test.go]: I believe it is addressed, I extended the existing test to perform the conversion in both directions and compare the result with the original Path - see: https://github.com/cilium/cilium/blob/7fb8fb5dbc342014b77dfec5e5549aaca10a5472/pkg/bgpv1/gobgp/conversions_test.go#L48-L52

**Comment by rastislavs** [comment]
@YutaroHayakawa I consolidated the commits

File: pkg/bgpv1/gobgp/conversions_test.go (hunk 1 above)
Comment (YutaroHayakawa) [reply pkg/bgpv1/gobgp/conversions_test.go]: Oh, sorry. I have missed that.

**Review by YutaroHayakawa — APPROVED** [review]
Now LGTM 👍 Thanks!

**Pull request merged**